integration_tests/contracts/contracts/ICA.sol

```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+interface IICAModule {
+    event SubmitMsgsResult(string seq);
+    function registerAccount(string calldata connectionID, string calldata version) external payable returns (bool);
+    function queryAccount(string calldata connectionID, address addr) external view returns (string memory);
+    function submitMsgs(string calldata connectionID, string calldata data, uint256 timeout) external payable returns (uint64);
```

Comment:
Here we are still missing a function to `queryPacketSeqResult(string seq) returns (Result)`

The usage is for the smart contract to read the result of previously sent packet sequence and execute subsequent actions if applicable.

Reply:
Yea, could add one if need

Reply:
Seems we need icaauth change for querySubmitMsgsResult, not sure if we can do in another PR

Reply:
It's not ideal to save the acknowledge in state, it bloat the state, and we might need to clean it up later, we can try callbacks.

Reply:
How does callback works?

> not sure if we can do in another PR
Yes as this PR has certain size already, if you feel like we want to do a cut-off here it's totally fine.

Reply:
we could try call back precompile in middleware to avoid need save extra data

Reply:
> How does callback works?

call contract method in icaauth ibc module, `OnAcknowledgement` and `OnTimeout`, the contract address is the port-id.

Reply:
Is it the same as this ADR?
https://github.com/cosmos/ibc-go/blob/v7.3.0/docs/architecture/adr-008-app-caller-cbs/adr-008-app-caller-cbs.md

Reply:
yeah, didn't know about this adr, [v7.3.0](https://github.com/cosmos/ibc-go/releases/tag/v7.3.0) seems perfect for us.

Reply:
I suggest handle the ack part in a separate issue.